pkg/v20/resource/appmigration/create.go

```diff
@@ -42,6 +47,99 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
+	clusterConfig, err := r.getClusterConfigFunc(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
```

Comment:
This is the migration logic. It's quite big but hopefully it's understandable. Process is

- Cordon chartconfig CR to prevent changes.
- Wait for app CR with a helm release status of deployed.
- Add the delete-custom-resource-only annotation.
- Delete the chartconfig CR.